app/Http/Controllers/SurveyController.php

```diff
@@ -716,10 +716,15 @@ public function storeResult(ResultRequest $request)
         DB::beginTransaction();
 
         try {
+            $survey = $this->surveyRepository->getSurveyFromToken($request->json()->get('survey_token'));
             $this->resultRepository->storeResult($request->json(), $this->surveyRepository);
             $request->session()->forget('current_section_survey');
 
-            DB::commit();
+            if (Carbon::now()->gt(Carbon::parse($survey->end_time)) || Carbon::now()->lt($survey->start_time)) {
+                DB::rollback();
+            } else {
+                DB::commit();
+            }
```

Comment:
Làm vậy thì nó update vô database xong hết rồi mới check rollback à ?
thử làm thế này nhé. Vô ajax js check có biến redirect thì chuyển hướng qua.

```suggestion
        try {
            $survey = $this->surveyRepository->getSurveyFromToken($request->json()->get('survey_token'));

            if (Carbon::now()->gt(Carbon::parse($survey->end_time)) || Carbon::now()->lt(Carbon::parse($survey->start_time))) {
                $redirect = route('survey.create.complete', $survey->token);
                throw new Exception("abc...xyz");
            } 

            $this->resultRepository->storeResult($request->json(), $this->surveyRepository);
            $request->session()->forget('current_section_survey');
            DB::commit();
             
            return response()->json([
                'success' => true,
            ]);
         } catch (Exception $e) {
             DB::rollback();
 
             return response()->json([
                 'success' => false,
                 'message' => trans('lang.send_result_failed'),
                 'redirect' => isset(redirect) ? $redirect : null,
             ]);
         }
```
```